pyrdp/layer/rdp/security.py

```diff
@@ -135,7 +135,9 @@ def __init__(self, parser = BasicSecurityParser()):
         self.securityHeaderExpected = False
 
     def recv(self, data: bytes):
-        if not self.securityHeaderExpected:
+        # Licensing happens in the security layer
+        licensingBytes = b"\x80\x00"
```

Comment:
Does that correspond to a flag?

Reply:
It is, but since this isn't parsed data, it is not in the correct format. Would it be better if I refer to this flag, and parse the 0x80\x00 and compare to the flag?

In pyrdp/enum/rdp.py - line 43
SEC_LICENSE_PKT = 0x0080

Reply:
Yeah for that @xshill would have a better idea than me. I have an idea but i’m really not sure if its good

Reply:
Reference: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpbcgr/e13405c5-668b-4716-94b2-1c2654ca1ad4

Reply:
The licensingBytes you're looking for IS part of a security header. You shouldn't need to check for that if securityHeaderExpected is set properly.

Reply:
Licensing is an optional part of the protocol, and the boolean "securityHeaderExpected" is false when receiving a licensing PDU. securityHeaderExpected is only used to redirect traffic to the securityLayer when we know we are expecting it in the handshake

Reply:
Oh right, since this is TLS it starts at false.